clang/docs/checkedc/rfc/Include-Checked-Hdrs.md

```diff
@@ -0,0 +1,187 @@
+          commandline.  Therefore specifying this flag will cause the implicit
+          inclusion of the checked counterpart of `foo.h`, which will make it
+          convenient for automation. Not specifying the commandline flag will
+          not perform implicit inclusion, satisfying our "opt-in" philosophy.
+          Infinite recursion is avoided because of `#include_next`.
+        - If a program includes `foo_checked.h` current behavior will prevail.
+          The above flag has no effect on the explicit inclusion of checked
+          header files.
+
+     - **Pros**: 1) It satisfies the "opt-in" philosophy. 2) It is convenient
+       for automation. 3) It provides fine-grained control to directly include a
+       system header file. 4) The existing approach of explicit inclusion is
+       supported. 5) In most cases integration with the build system is easy:
+       the flag `-DIMPLICIT_INCLUDE_CHECKED_HDRS` is passed to the compilation
+       of all source files through the `CFLAGS` variable.
+
+     - **Cons**: 1) In some cases integration with the build system will be more
+       involved if the above flag needs to be passed to the compilation of most
+       source files and avoided for a few source files that may want to directly
+       include system header files in order to avoid Checked-C-specific
+       declarations. 2) It is difficult to accomodate clang-specific
+       declarations.
+
+## Proposed Solution:
+
+The proposed solution is as follows:
+  - It is a hybrid solution based on ideas in Solutions 3 and 1 that
+    handles the presence of both clang-specific and Checked-C-specific
+    declarations in header files. It also incorporates the feedback from CCI to
+    make implicit inclusion of checked header files as default.
+  - We primarily adopt Solution 3 for header files that do not have
+    clang-specific declarations, but have Checked-C-specific declarations (there
+    are 13 such header files at present). To implicitly include checked headers
+    by default, the compilation flag (as proposed in Solution 3) now gives a way
+    to **opt out** of implicit inclusion. Accordingly, it is renamed to
+    NO_IMPLICIT_INCLUDE_CHECKED_HDRS.
+      - For each system header file, say `foo.h`, that does not have
+        clang-specific declarations but has Checked-C-specific declarations, we
+        will add a new file also called `foo.h` that will contain the following:
+
+            #ifdef NO_IMPLICIT_INCLUDE_CHECKED_HDRS
+            #include_next <foo.h>
+            #else
+            #include <foo_checked.h>
+            #endif
+
+      - The file `foo_checked.h` will contain `#include_next <foo.h>` and the
+        Checked-C-specific declarations.
+
+  - For a system header file that contains clang-specific declarations and also
+    Checked-C-specific declarations (there is one such header file at present),
+    we will do the following:
```

Comment:
If you're willing to change the default include path in the driver to add a Checked-C-specific include directory (say `lib/clang/VERSION/include-checkedc`) before the main one (`lib/clang/VERSION/include`), then you won't need to do this.  You can use the solution in the previous bullet point for all Checked C headers, and the `#include_next <foo.h>` will automatically pick up either the system header or the Clang header (in which the `#include_next <foo.h>` would then include the system header) as appropriate.  This would also cleanly separate the Checked C headers from the Clang headers in different subdirectories of the installation directory, making it clearer for users during troubleshooting.

Reply:
We have decided not to change the default include path in the driver at present.